Assets/Source/Components/Enemy/SimpleEnemyBehavior.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using Assets.Source.Components.Enemy.Base;
+using Assets.Source.Components.Reactor.Interfaces;
+using Assets.Source.Constants;
+using Assets.Source.Extensions;
+using UnityEngine;
+
+namespace Assets.Source.Components.Enemy
+{
+    /// <summary>
+    /// This enemy is basic enemy that will shoot at the player when close enough and shoots at a regular interval,
+    /// it also gets stunned when hit by a projectile and has relatively low health.
+    /// </summary>
+    public class SimpleEnemyBehavior : EnemyAIBase, IProjectileReactor
+    {
+        private readonly float MOVE_SPEED = 2.0f;
+        private readonly float SHOOT_COOLDOWN = 1.5f; //the amount of time allowed between firing shots
```

Comment:
I like this.  Eventually the player will do that same thing though so we will want to move that functionality over to the actor behavior or something

Reply:
Yeah sounds good to me, that should be easy enough to do when we are ready for that. I did think about putting it in the base actor behavior but then would it be like a virtual member so that way you can change it between various enemies and the player if you like? I also enjoy having the player being able to shoot super fast right now lol